### PR TITLE
Fix data/plugin folders mounted as root

### LINKED ISSUE #129: Unable to update to newest version: Problem with file storage settings
After an update from v5.13.2 to v5.17.1 (mattermost-team-edition) (official helm chart kubernetes deployment) our mattermost instance throws the following error while starting:

`{"level":"error","ts":1575923795.6970422,"caller":"app/server_app_adapters.go:134","msg":"Problem with file storage settings: TestFileConnection: api.file.test_connection.local.connection.app_error, WriteFile: api.file.write_file_locally.writing.app_error, open data/testfile: permission denied"}`

The instance goes online nevertheless but all media files in all channels are missing now.
A rollback fixes this problem.

What do we have to additionally consider to be able to update our instance to the newest available version?

### PR DESCRIPTION
**Pull request by etlfg**
Due to mattermost image needs mattermost ownership for version >= 5.14

#### Summary

Mount volumes as mattermost user if mattermost image is greater than 5.14 (first version to use mattermost user to run processes in the container)

#### Ticket Link

Fixes #129, #139, #142

### CODE REVIEW

File: charts/mattermost-team-edition/templates/deployment.yaml [hunk 1]
```diff
@@ -80,6 +80,12 @@ spec:
         {{- end }}
         resources:
           {{- .Values.resources | toYaml | nindent 12 }}
+      {{ if and (semverCompare ">= 5.14" .Values.image.tag) (or .Values.persistence.data.enabled .Values.persistence.plugins.enabled) }}
+      securityContext:
```
Comment (cpanato) [inline charts/mattermost-team-edition/templates/deployment.yaml:84]: @etlfg thanks for this PR
can you make this customized? and add those in the values.yaml and template here?

thanks

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks for opening this PR!
a small request

File: charts/mattermost-team-edition/templates/deployment.yaml (hunk 1 above)
Comment (etlfg) [reply charts/mattermost-team-edition/templates/deployment.yaml:84]: What do you mean by making it customized ? The version being compared to the image tag (needs to be an hardcoded value to me) or the user being defined in the securityContext ?

Comment (cpanato) [reply charts/mattermost-team-edition/templates/deployment.yaml:84]: if someone builds a custom image and change the user in the image this chart will not work for them

Comment (etlfg) [reply charts/mattermost-team-edition/templates/deployment.yaml:84]: Got it. CI is being running on my proposal concerning your request.

**Comment by jfrerich** [comment]
@cpanato, @stylianosrigas ping for review

**Comment by JasonGilholme** [comment]
Could this get merged please?

I was going to submit the same PR as I had to spend some time sorting out the init container hack today.  It would be great if I could remove my dirty work!!

Thanks in advance

**Comment by cpanato** [comment]
@etlfg can you rebase and fix the conflict so we can merge it?

**Comment by debarshibasak** [comment]
Could this be merged?

**Comment by stafot** [comment]
@debarshibasak It has conflicts with the `master` branch. It needs to be rebased.

**Comment by jasonblais** [comment]
@metanerd I helped resolve the merge conflicts (though looks like the tests failed, not sure why). 

Queuing for your review next, or please feel free to re-assign to another member of your team.

**Comment by emilyacook** [comment]
> @metanerd I helped resolve the merge conflicts (though looks like the tests failed, not sure why).
> 
> Queuing for your review next, or please feel free to re-assign to another member of your team.

@metanerd gentle ping on this!

**Review by khos2ow — COMMENTED** [review]
Generally this LGTM, but our testing infra was quite outdated and has been fixed in #205. When rebased on master one more time, this should be ok.

**Comment by lindy65** [comment]
Hi @etlfg - I'm helping to review stale PRs and, since we haven't heard from you in a few months, I'll close this PR for now. Please feel free to re-open if you're able to prioritize. Thanks!

//cc @cpanato @metanerd

**Pull request closed without merging**